### DISCUSSION TITLE
How to persist filter params after running an action on in a has_many_field turbo frame?

### DISCUSSION THREAD

**Discussion by alimi** [avo-hq/avo#3463] [Help]
On a show page with a has_many association, you can filter the has_many records. For example if a company has_many customers and I'm on the company page with a `has_many :customers` field, I could filter the customers by name. Then, I might run an action on the filtered customers. When I finish running the action, the page reloads and the customers filters are reset.

I believe the filters get reset because `Avo::ActionsController` is redirecting to the request referrer.

https://github.com/avo-hq/avo/blob/c60e8543806981d2378cec5e1375b247af03ead6/app/controllers/avo/actions_controller.rb#L131-L135

The filter params would be in the `has_many_field_show_customers` turbo frame as the src, but the request referrer is `/resources/company/42` which has no information about the turbo frame. If we could pass the turbo frame information to the action, could `Avo::ActionsController#respond` refresh the turbo frame instead of reloading the whole page? Then, the turbo frame would have its filters in place.

Or is there another way to keep the filter params around? In our app we do a lot of filtering and data updates, and losing the filters is a bummer.

**Comment by Paul-Bob** [comment]
Hello.

You could use `close_modal` (aliased as `do_nothing`) https://docs.avohq.io/3.0/actions.html#close_modal and it will just close the modal and flash the alerts without any redirect.

If you want to reload the table content you could use `append_to_response` (https://docs.avohq.io/3.0/actions.html#append_to_response), inside that block you can use the `turbo_frame_reload`  to reload the frame around the has many field.

Something like:

```ruby
def handle(**args)
  succeed "Modal closed!!"
  close_modal

  append_to_response -> {
    turbo_stream.turbo_frame_reload(the_turbo_frame)
  }
end
```

Let us know how it goes

**Reply by alimi** [reply]
Awesome, one of those should work! I’ll follow up when I’m back at my computer next week.
